src/mmda/utils/tools.py

```diff
@@ -41,20 +41,33 @@ def allocate_overlapping_tokens_for_box(
 
 
 def box_groups_to_span_groups(
-        box_groups: List[BoxGroup], doc: Document, pad_x: bool = False, center: bool = False
-) -> List[SpanGroup]:
-    """Generate SpanGroups from BoxGroups.
+        box_groups: List[BoxGroup],
+        doc,
+        pad_x: bool = False,
+        center: bool = False,
+        unallocated_tokens_dict:  Optional[Dict[int, SpanGroup]] = None
+) -> Union[List[SpanGroup], Tuple[List[SpanGroup], Dict[int, SpanGroup]]]:
```

Comment:
I don't think you need the union-type return. You can always just return `List[SpanGroup]`. Since you're mutating `unallocated_tokens_dict` when it's passed in, the caller will always have the latest.